packages/server/src/api/controllers/static/index.ts

```diff
@@ -269,9 +270,14 @@ export const getSignedUploadURL = async function (ctx: Ctx) {
   const awsRegion = (datasource?.config?.region || "eu-west-1") as string
   if (datasource?.source === "S3") {
     const { bucket, key } = ctx.request.body || {}
+
+    // Ensure we aren't using a custom endpoint
+    if (datasource?.config?.endpoint) {
+      ctx.throw(400, "S3 datasources with custom endpoints are not supported")
```

Comment:
How come?

Reply:
I don't believe our infrastructure can currently support custom endpoints. There would have to be a fairly big update to support file storage and retrieval in the builder. Definitely possible though.